x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_helpers.ts

```diff
@@ -181,6 +201,60 @@ export function getVisualizeFieldSuggestions({
     visualizationState: undefined,
     visualizeTriggerFieldContext,
   });
+
+  if (visualizeTriggerFieldContext && 'isVisualizeAction' in visualizeTriggerFieldContext) {
+    const { layers, configuration } = visualizeTriggerFieldContext;
+    const allSuggestions = suggestions.filter(
+      (s) => s.visualizationId === activeVisualization?.id
```

Comment:
It seems like this kind of working accidentally right now because the default active visualization is XY. What about making the target vis type part of the passed in context? It will couple the caller (TSVB, later Visualize and so on) to the Lens visualization types but that doesn't seem like a problem